### PR TITLE
.travis.yml: simplify build commands

### PR DESCRIPTION
**Pull request by jforissier**
Many build commands needlessly set some variables to their default
values: simplify that. Also, reformat commands to make lines shorter,
and introduce variable make="make -j3 -s".

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: .travis.yml [hunk 1]
```diff
@@ -73,79 +78,78 @@ script:
   - git format-patch -1 --stdout | $DST_KERNEL/scripts/checkpatch.pl --ignore FILE_PATH_CHANGES --ignore GERRIT_CHANGE_ID --no-tree -
 
   # Orly2
-  -                                  PLATFORM=stm-orly2                                  CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1 PLATFORM=stm                PLATFORM_FLAVOR=orly2   CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=0 DEBUG=0 PLATFORM=stm                PLATFORM_FLAVOR=orly2   CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
+  - $make PLATFORM=stm-orly2
+  - $make PLATFORM=stm-orly2 CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1
+  - $make PLATFORM=stm-orly2 CFG_TEE_CORE_LOG_LEVEL=0
```
Comment (pascal-brand-st-dev) [inline .travis.yml]: Would like at least one compilation with "PLATFORM=stm PLATFORM_FLAVOR=orly2" to check we can pass it that way.

File: .travis.yml [hunk 2]
```diff
@@ -73,79 +78,78 @@ script:
...
 
   # Cannes
-  -                                  PLATFORM=stm-cannes                                 CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1 PLATFORM=stm                PLATFORM_FLAVOR=cannes  CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=0 DEBUG=0 PLATFORM=stm                PLATFORM_FLAVOR=cannes  CROSS_COMPILE=arm-linux-gnueabihf-  make -j8 all -s
+  - $make PLATFORM=stm-cannes
+  - $make PLATFORM=stm-cannes CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1
+  - $make PLATFORM=stm-cannes CFG_TEE_CORE_LOG_LEVEL=0
```
Comment (pascal-brand-st-dev) [inline .travis.yml]: DEBUG is not 0 by default, it is undefined

File: .travis.yml [hunk 3]
```diff
@@ -73,79 +78,78 @@ script:
...
 
   # FVP
-  -                                  PLATFORM=vexpress-fvp       CFG_ARM32_core=y   CROSS_COMPILE_core=arm-linux-gnueabihf- make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1 PLATFORM=vexpress-fvp       CFG_ARM32_core=y   CROSS_COMPILE_core=arm-linux-gnueabihf- make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=0 DEBUG=0 PLATFORM=vexpress-fvp       CFG_ARM32_core=y   CROSS_COMPILE_core=arm-linux-gnueabihf- make -j8 all -s
-  -                                  PLATFORM=vexpress-fvp       CFG_ARM64_core=y   CROSS_COMPILE_core=aarch64-linux-gnu-   CROSS_COMPILE_ta_arm64=aarch64-linux-gnu- make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=4 DEBUG=1 PLATFORM=vexpress-fvp       CFG_ARM64_core=y   CROSS_COMPILE_core=aarch64-linux-gnu-   CROSS_COMPILE_ta_arm64=aarch64-linux-gnu- make -j8 all -s
-  - CFG_TEE_CORE_LOG_LEVEL=0 DEBUG=0 PLATFORM=vexpress-fvp       CFG_ARM64_core=y   CROSS_COMPILE_core=aarch64-linux-gnu-   CROSS_COMPILE_ta_arm64=aarch64-linux-gnu- make -j8 all -s
+  - $make PLATFORM=vexpress-fvp
```
Comment (pascal-brand-st-dev) [inline .travis.yml]: It does not hurt to have, in one build, `CFG_ARM32_core=y`

**Comment by pascal-brand-st-dev** [comment]
With or without my comments:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

File: .travis.yml (hunk 1 above)
Comment (jforissier) [inline .travis.yml]: OK

File: .travis.yml (hunk 2 above)
Comment (jforissier) [inline .travis.yml]: Yes, but anything other than `1` means not debug. Anyway I will add `DEBUG=0` once.

File: .travis.yml (hunk 3 above)
Comment (jforissier) [inline .travis.yml]: OK

**Comment by jforissier** [comment]
Comments addressed and tags applied.

**Pull request merged**